### PR TITLE
Add transform for wrw

### PR DESCRIPTION
**Pull request by zjing14**

### CODE REVIEW

File: miopen_dialect_idea.md [hunk 1]
```diff
@@ -380,6 +380,181 @@ miopen.gridwise_gemm(%filter_gemmK_gemmM, %output_gemmK_gemmN, %input_gemmM_gemm
 
 -------------------------------------------------------------------------------
 
+conv backward weight transform
+===============
+
+an example based on nchw/kcyx/nkhw:
```
Comment (jerryyin) [inline miopen_dialect_idea.md:386]: Nit: Capitalize the first letter to be consistent with backward data?

File: miopen_dialect_idea.md [hunk 2]
```diff
@@ -380,6 +380,181 @@ miopen.gridwise_gemm(%filter_gemmK_gemmM, %output_gemmK_gemmN, %input_gemmM_gemm
...
+      dimensions = [0],
+      names = ["gemmM"],
+      transformation = "PassThrough",
+      source_dimensions = [0],
+      source_names = ["k"]
+    },
+    {
+      dimensions = [1],
+      names = ["gemmN"],
+      transformation = "Unfold",
```
Comment (jerryyin) [inline miopen_dialect_idea.md:403]: Just to clarify, "Unfold" only applies to KCYX layout, but not, for example CKYX? (which would apply for "Merge")

**Review by jerryyin — APPROVED** [review]
Looking good to me. Good job!

Only one Nit comment, leaving it to you to decide fix/not.

File: miopen_dialect_idea.md (hunk 2 above)
Comment (whchung) [reply miopen_dialect_idea.md:403]: unfold should only apply to consecutive source dimensions.

**Review by whchung — APPROVED** [review]

**Pull request merged**